### DISCUSSION TITLE
What is the proven max number of nodes per node pool?

### DISCUSSION THREAD

**Discussion by maggie44** [kube-hetzner/terraform-hcloud-kube-hetzner#709] [Q&A] [answered]
In the README it notes:

> As for the count of nodes per nodepools, if you raise your limits in Hetzner, you can have up to 64,670 nodes per nodepool (also very unlikely to need that much).

What is this based on? Is this PIDs? Has it been tested? And what is meant by raising the limits at Hetzner, like done on the host with ulimits?

I figured there were limits hit well before that, maybe related to bridge network limits for example? I am curious how many nodes people have been able to achieve.

**Comment by mysticaltech** [comment, marked as answer]
@maggie44 This is based on IPs calculations because we have to pre-allocate the private network IPs. Now if you try with a new Hetzner account to deploy a hundred nodes, you will get an error, stating you need to contact Hetzner to require permission for more nodes, this is out of extra precaution from them to make sure that you can afford such a setup based on your account history. 

Other than that, if you want to deploy more than 500 nodes, you need to disable placement groups, we have a variable for that, because they only support up 50 placement groups of 10 nodes per network. 

When you do that, there is not upper limit, k3s can handle an infinite number of nodes, but you will need more powerful control planes of course to handle all these agents, or scale them horizontally too.
